server/websocket/src/server.rs

```diff
@@ -137,19 +158,30 @@ where
                             let service = GraphQlConnection::new(
                                 &logger,
                                 subgraphs.clone(),
-                                subgraph,
+                                subgraph.clone(),
                                 ws_stream,
                                 graphql_runner.clone(),
                             );
-                            tokio::spawn(service.into_future()).into_future()
+
+                            let cancel_subgraph = subgraph.clone();
+                            let (service, guard) = service.into_future().cancelable(move || {
```

Comment:
I know it was called `service` before (my bad), but can we rename it to `conn` or `gql_conn`?